[debug] [2018-12-13T09:12:49.840Z] ----------------------------------------------------------------------
[debug] [2018-12-13T09:12:49.843Z] Command:       /usr/local/bin/node /usr/local/bin/firebase serve
[debug] [2018-12-13T09:12:49.843Z] CLI Version:   5.1.1
[debug] [2018-12-13T09:12:49.843Z] Platform:      darwin
[debug] [2018-12-13T09:12:49.843Z] Node Version:  v8.11.3
[debug] [2018-12-13T09:12:49.843Z] Time:          Thu Dec 13 2018 17:12:49 GMT+0800 (AWST)
[debug] [2018-12-13T09:12:49.843Z] ----------------------------------------------------------------------
[debug] 
[debug] [2018-12-13T09:12:49.861Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2018-12-13T09:12:49.861Z] > authorizing via signed-in user
[debug] [2018-12-13T09:12:49.862Z] [iam] checking project coki-214004 for permissions ["firebase.projects.get"]
[debug] [2018-12-13T09:12:49.863Z] > refreshing access token with scopes: ["email","https://www.googleapis.com/auth/cloud-platform","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","openid"]
[debug] [2018-12-13T09:12:49.863Z] >>> HTTP REQUEST POST https://www.googleapis.com/oauth2/v3/token  
 <request body omitted>
[debug] [2018-12-13T09:12:50.544Z] <<< HTTP RESPONSE 200
[debug] [2018-12-13T09:12:50.596Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/coki-214004:testIamPermissions  

[debug] [2018-12-13T09:12:52.599Z] <<< HTTP RESPONSE 200
[debug] [2018-12-13T09:12:52.600Z] >>> HTTP REQUEST GET https://cloudresourcemanager.googleapis.com/v1/projects/coki-214004  
 
[debug] [2018-12-13T09:12:53.973Z] <<< HTTP RESPONSE 200
[info] 
[info] === Serving from '/Users/277594b/Documents/repos/coki-firebase'...
[info] 
[debug] [2018-12-13T09:12:54.861Z] Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/logs'
    at Object.fs.mkdirSync (fs.js:885:18)
    at Object.assertLogsPath (/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/src/emulator/logs.js:65:10)
    at new Controller (/usr/local/lib/node_modules/firebase-tools/node_modules/@google-cloud/functions-emulator/src/cli/controller.js:77:32)
    at FunctionsEmulator.start (/usr/local/lib/node_modules/firebase-tools/lib/functionsEmulator.js:113:21)
    at Object.start (/usr/local/lib/node_modules/firebase-tools/lib/serve/functions.js:9:29)
    at /usr/local/lib/node_modules/firebase-tools/lib/serve/index.js:17:21
    at arrayMap (/usr/local/lib/node_modules/firebase-tools/node_modules/lodash/lodash.js:639:23)
    at Function.map (/usr/local/lib/node_modules/firebase-tools/node_modules/lodash/lodash.js:9556:14)
    at _serve (/usr/local/lib/node_modules/firebase-tools/lib/serve/index.js:15:7)
    at Command._action (/usr/local/lib/node_modules/firebase-tools/commands/serve.js:47:12)
[error] 
[error] Error: An unexpected error has occurred.
